main.c

```diff
@@ -65,12 +66,25 @@ int main(void) {
     txb_init(tx_pool, sizeof(tx_pool), can_send, can_send_rdy);
 
     uint32_t last_millis = millis();
+    uint32_t last_message_millis = millis();
 
     // Turn LED 1 on
     LED_1_ON();
 
     // Wait for the first message
     while (!recieved_first_message) {
+        CLRWDT(); // feed the watchdog, which is set for 256ms
+        
+        if (seen_can_message) {
+            seen_can_message = false;
+            last_message_millis = millis();
+        }
+        
+        if (millis() - last_message_millis > MAX_BUS_DEAD_TIME_ms) {
+            // We've got too long without seeing a valid CAN message (including one of ours)
+            RESET();
+        }
+        
         if (millis() - last_millis > 5000) {
```

Comment:
Yes, since if we're not seeing our own can messages we have a bigger problem. Although I notice we never send a status message while in this loop? So the board would appear dead to the operators?

Reply:
IMO this is a github issue + merge though, we shouldn't block these PRs.

Reply:
sgtm